src/osgPlugins/dds/ReaderWriterDDS.cpp

```diff
@@ -1031,12 +1031,15 @@ osg::Image* ReadDDSFile(std::istream& _istream, bool flipDDSRead)
     {
         // Now we need to substitute the indexed image data with full RGBA image data.
         unsigned char* convertedData = new unsigned char [sizeWithMipmaps * 4];
+        unsigned char* pconvertedData = convertedData;
+        unsigned char *pimageData = imageData;
+        unsigned char *ppalet;
         for (unsigned int i = 0; i < sizeWithMipmaps; i++)
         {
-            convertedData[i * 4 + 0] = palette[imageData[i] * 4 + 0];
-            convertedData[i * 4 + 1] = palette[imageData[i] * 4 + 1];
-            convertedData[i * 4 + 2] = palette[imageData[i] * 4 + 2];
-            convertedData[i * 4 + 3] = palette[imageData[i] * 4 + 3];
+            ppalet = &palette[(*pimageData)*4];
+            memcpy(pconvertedData, ppalet, 4);
```

Comment:
memcpy can be replace with 
`                   *((unsigned int *)pconvertedData)=*((unsigned int *)(ppalet));`
but it's not very pretty

Reply:
Then I propose [this](https://github.com/Capostrophic/OpenSceneGraph/commit/028c7edf8bf8304fe448606ca898cb6dce788ee9).